modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/filename/FileTreeUtils.java

```diff
@@ -61,6 +63,20 @@ public static void removeTmpSnapshotFiles(SnapshotFileTree sft, boolean err, Ign
             removeTmpDir(tmpDrStorage.getParentFile(), err, log);
     }
 
+    /**
+     * @param ccfg Cache configuration.
+     * @param part Partition.
+     * @return Storage path from config for partition.
+     */
+    public static @Nullable String partitionStorage(CacheConfiguration<?, ?> ccfg, int part) {
+        String[] csp = ccfg.getStoragePaths();
+
+        if (csp == null)
```

Comment:
I would add an array empty check to make sure we won't perform division by zero down the code flow.

Reply:
Done